### PR TITLE
[5.x] feature: this adds the vite integrity plugin to support assetsAreCurrent check

### PR DESCRIPTION
**Pull request by mmachatschek**
This adds the manifest integrity plugin for vite to support checks if the user published the current version of the horizon assets.

### CODE REVIEW

File: resources/views/layout.blade.php [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+
+$viteDataSchemeLight = new ViteFoundation();
+$viteDataSchemeLight->useStyleTagAttributes([
+'data-scheme' => 'light',
+]);
+
+$viteDataSchemeDark = new ViteFoundation();
+$viteDataSchemeDark->useStyleTagAttributes([
+'data-scheme' => 'dark',
+]);
```
Comment (mmachatschek) [inline resources/views/layout.blade.php:13]: if there is a better solution to this please tell me. currently its fairly hard (impossible) to add these kind of style tag attributes with the @vite blade directive

File: resources/views/layout.blade.php [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
 @endphp
 <!DOCTYPE html>
 <html lang="en">
+
```
Comment (driesvints) [inline resources/views/layout.blade.php]:
```suggestion
```

**Comment by driesvints** [comment]
@mmachatschek what would they see if the assets aren't up to date? I also thought we already had a check for this 🤔

**Comment by mmachatschek** [comment]
@driesvints there is a file compare check here https://github.com/laravel/horizon/blob/3c6661d19013d108143ac204bc5f0e4f3313bf68/src/Horizon.php#L195 that checks if the manifest.json in the published folder is the same as the manifest.json from the package.

with laravel mix we used versioning that added a hash `?id=` for invalidating https://github.com/laravel/horizon/blob/32ca28dcb91fa2f4fa650f21440144c4bc2846dd/public/mix-manifest.json#L2 the built files.

the integrity package readds this missing feature after my change from mix to vite

**Review by driesvints — APPROVED** [review]

**Pull request merged**